04.ls/ls.rb

```diff
@@ -2,26 +2,126 @@
 # frozen_string_literal: true
 
 require 'optparse'
+require 'etc'
 
 COLUMNS = 3
 TAB_WIDTH = 8
 
+ENTRY_TYPES = {
+  file: '-',
+  blockSpecial: 'b',
+  characterSpecial: 'c',
+  directory: 'd',
+  link: 'l',
+  fifo: 'p',
+  socket: 's',
+  unknown: 'w'
+}.freeze
+
+FILE_PERMISSIONS = {
+  '0': '---',
+  '1': '--x',
+  '2': '-w-',
+  '3': '-wx',
+  '4': 'r--',
+  '5': 'r-x',
+  '6': 'rw-',
+  '7': 'rwx'
+}.freeze
+
+HALF_A_YEAR_SECONDS = (60 * 60 * 24) * (365.2425 / 2)
+
 def main
-  file_names = search_file_names
+  options = ARGV.getopts('alr')
+  searched_file_names = options['a'] ? Dir['..', '.*', '*'] : Dir['*']
+  sorted_file_names = searched_file_names.sort_by(&:downcase)
+  file_names = options['r'] ? sorted_file_names.reverse : sorted_file_names
+  options['l'] ? display_long_format(file_names) : display_column_format(file_names)
+end
+
+def display_long_format(file_names)
+  long_formats = build_long_format_table(file_names)
+  total_blocks = long_formats.sum { |format| format[:blocks] }
+  puts "total #{total_blocks}"
+
+  widths = caluculate_max_length(long_formats)
+  long_formats.each do |format|
+    rows = [
+      "#{format[:file_mode]} ",
+      format[:hard_links].rjust(widths[:link_width]),
```

Comment:
```suggestion
      format[:hard_links].rjust(widths[:hard_links]),
```

widthは変数名から伝わるので、キー名を揃えてあげるとか

Reply:
キー名を統一しました